tests/aws/services/cloudformation/api/test_stacks.py

```diff
@@ -717,6 +715,37 @@ def test_list_parameter_type(aws_client, deploy_cfn_template, cleanups):
     assert stack.outputs["ParamValue"] == "foo|bar"
 
 
+@markers.aws.validated
+def test_subnet_id_parameter_type(
+    aws_client,
+    deploy_cfn_template,
+    cleanups,
+):
+    vpc_id = aws_client.ec2.create_vpc(CidrBlock="10.0.0.0/16")["Vpc"]["VpcId"]
+    cleanups.append(lambda: aws_client.ec2.delete_vpc(VpcId=vpc_id))
+    aws_client.ec2.get_waiter("vpc_available").wait(VpcIds=[vpc_id])
+    subnet_id_1 = aws_client.ec2.create_subnet(VpcId=vpc_id, CidrBlock="10.0.0.0/24")["Subnet"][
+        "SubnetId"
+    ]
+    cleanups.append(lambda: aws_client.ec2.delete_subnet(SubnetId=subnet_id_1))
+    subnet_id_2 = aws_client.ec2.create_subnet(VpcId=vpc_id, CidrBlock="10.0.1.0/24")["Subnet"][
+        "SubnetId"
+    ]
+    cleanups.append(lambda: aws_client.ec2.delete_subnet(SubnetId=subnet_id_2))
+
+    subnets_list = ",".join([subnet_id_1, subnet_id_2])
+    stack = deploy_cfn_template(
+        template_path=os.path.join(
+            os.path.dirname(__file__), "../../../templates/cfn_parameter_list_subnet_id_type.yaml"
+        ),
+        parameters={
+            "ParamsList": subnets_list,
+        },
+    )
+
+    assert stack.outputs["ParamValue"] == subnets_list.replace(",", "|")
```

Comment:
Why not a snapshot? The assertion is still the same but snapshots also provide nice docs for the real outputs for learning etc.

Not a blocker